packages/stryker/src/logging/LogConfigurator.ts

```diff
@@ -0,0 +1,138 @@
+  color: {
+    type: 'pattern',
+    pattern: '%[%r (%z) %p %c%] %m'
+  },
+  noColor: {
+    type: 'pattern',
+    pattern: '%r (%z) %p %c %m'
+  }
+};
+
+interface AppendersConfiguration {
+  [name: string]: log4js.Appender;
+}
+
+const LOG_FILE_NAME = 'stryker.log';
+export default class LogConfigurator {
+
+  private static createMasterAppenders(consoleLogLevel: LogLevel, fileLogLevel: LogLevel): AppendersConfiguration {
+
+    const multiAppender = { type: require.resolve('./MultiAppender'), appenders: ['filteredConsole'] };
+    let allAppenders: AppendersConfiguration = {
+      [AppenderName.Console]: { type: 'stdout', layout: layouts.color },
+      [AppenderName.FilteredConsole]: { type: 'logLevelFilter', appender: 'console', level: consoleLogLevel },
+      [AppenderName.All]: multiAppender,
+    };
+
+    // only add file if it is needed. Otherwise log4js will create the file directly, pretty annoying.
+    if (fileLogLevel.toUpperCase() !== LogLevel.Off.toUpperCase()) {
+      const fileAppender: log4js.FileAppender = { type: 'file', filename: LOG_FILE_NAME, layout: layouts.noColor };
+      const filteredFileAppender: log4js.LogLevelFilterAppender = { type: 'logLevelFilter', appender: 'file', level: fileLogLevel };
+
+      // Don't simply add the appenders, instead actually make sure they are ordinal "before" the others.
+      // See https://github.com/log4js-node/log4js-node/issues/746
+      allAppenders = Object.assign({ [AppenderName.File]: fileAppender, [AppenderName.FilteredFile]: filteredFileAppender }, allAppenders);
+
+      multiAppender.appenders.push(AppenderName.FilteredFile);
+    }
+
+    return allAppenders;
+  }
+
+  private static createLog4jsConfig(defaultLogLevel: LogLevel, appenders: AppendersConfiguration): log4js.Configuration {
+    return {
+      appenders,
+      categories: {
+        default: {
+          appenders: [AppenderName.All], level: defaultLogLevel
+        }
+      }
+    };
+  }
+
+  private static setImplementation(): void {
+    LoggerFactory.setLogImplementation(log4js.getLogger);
+  }
+
+  /**
+   * Configure logging for the master process. Either call this method or `forWorker` before any `getLogger` calls.
+   * @param consoleLogLevel The log level to configure for the console
+   * @param fileLogLevel The log level to configure for the "stryker.log" file
+   */
+  static forMaster(consoleLogLevel: LogLevel = LogLevel.Information, fileLogLevel: LogLevel = LogLevel.Off) {
+    this.setImplementation();
+    const appenders = this.createMasterAppenders(consoleLogLevel, fileLogLevel);
+    log4js.configure(this.createLog4jsConfig(minLevel(consoleLogLevel, fileLogLevel), appenders));
+  }
+
+  /**
+   * Configure the logging for the server. Includes the master configuration.
+   * This method can only be called ONCE, as it starts the log4js server to listen for log events.
+   * It returns the logging client context that should be used to doe the `forWorker` calls.
```

Comment:
What do you mean? There seem to be some errors in the last part of this sentence.

Reply:
Rephrased it as: ` It returns the logging client context that should be used to configure the child processes.`. I've also renamed the `forWorker` method to `configureChildProcess`. Is it clearer now?